packages/editor/src/components/rich-text/index.js

```diff
@@ -713,9 +713,14 @@ export class RichText extends Component {
 		if ( shouldReapply ) {
 			const record = this.formatToValue( value );
 
-			// Maintain the previous selection:
-			record.start = this.state.start;
-			record.end = this.state.end;
+			// If we always specify this the selection will go to this RichText
+			// regardless whether is is actually the currently selected one.
+			// So check if this RichText is the selected one.
+			if ( this.props.isSelected ) {
```

Comment:
@iseulde Any thoughts about this particular change?

Reply:
I've put some extra context about this change in the commit: https://github.com/WordPress/gutenberg/pull/12186/commits/50057fc750b33fd7ba99844e49c0c9dc1cf05152.